packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -610,6 +710,56 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   *   Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization
+  */
+  public enableDeploymentAlarms(alarmConfig: DeploymentAlarmConfig) {
+    // Throw an error if deployment alarms are already configured
+    if (this.deploymentAlarms) {
+      throw new Error('Deployment alarms are already configured.');
+    }
+    // Throw an error if alarms array is empty
+    if (alarmConfig.alarms.length === 0) {
+      throw new Error('Specify at least one deployment alarm');
+    }
+    this.deploymentAlarms = {
+      alarmNames: alarmConfig.alarms.map(alarm => alarm.alarmName),
+      enable: true,
+      rollback: alarmConfig.behavior !== AlarmBehavior.FAIL_ON_ALARM,
+    };
+  }
+
+  /**
+   *   Add an alarm based on ECS metric which appended to the list of deployment alarms at synthesis time
+   *   Customize the alarm threshold and evaluationPeriods using 2nd argument ecsMetricAlarmProps.
+   *   `threshold  value 85`
+   *   `evaluationPeriods value 3`
+  */
+  public createEcsMetricAlarm(metric: EcsMetric, ecsMetricAlarmProps?: EcsMetricAlarmProps): cloudwatch.Alarm {
```

Comment:
Are we using this anywhere?

Reply:
Yes we are using this for Customize the alarm threshold and evaluationPeriods using 2nd argument ecsMetricAlarmProps and I also added validations in it .
I added unit test cases for the validations present in it.

Reply:
I tried to search for `createEcsMetricAlarm` and can't find any reference. Can you paste a link where it get used?